### PR TITLE
Upgrade to Unicode v15.0.0

### PR DESCRIPTION
**Pull request by meowgorithm**
This upgrades to Unicode v15 and regenerates tables accordingly.

### CODE REVIEW

**Comment by rivo** [comment]
Thanks. There are still a few references to Unicode 14 in the code (and the README). Can you please update those, too?

Also, do you know if any of the rules have changed between 14 and 15? I suppose they haven't but I want to be sure.

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 [![Go Reference](https://pkg.go.dev/badge/github.com/rivo/uniseg.svg)](https://pkg.go.dev/github.com/rivo/uniseg)
 [![Go Report](https://img.shields.io/badge/go%20report-A%2B-brightgreen.svg)](https://goreportcard.com/report/github.com/rivo/uniseg)
 
-This Go package implements Unicode Text Segmentation according to [Unicode Standard Annex #29](https://unicode.org/reports/tr29/), Unicode Line Breaking according to [Unicode Standard Annex #14](https://unicode.org/reports/tr14/) (Unicode version 14.0.0), and monospace font string width calculation similar to [wcwidth](https://man7.org/linux/man-pages/man3/wcwidth.3.html).
```
Comment (rivo) [inline README.md:6]: This was not the Unicode version but the report number. Please change this back (and elsewhere where you did that). You can't just search+replace here, you'll have to look where the actual version number was mentioned.

**Review by rivo — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (meowgorithm) [reply README.md:6]: Sounds good and apologies, I'll give this another shot.

**Comment by meowgorithm** [comment]
With regard to changes in rules, I'm not aware of any, though I'm also not an expert here. As such, I do understand if you'd prefer err on the side of caution.

**Comment by meowgorithm** [comment]
So I believe this is ready now (apologies for the noise with the force pushes) however let me know if there's anything I missed. I'm not aware of any changes in rules that may or may not have occurred so it may be wise to approach this one cautiously as mentioned.

Regardless, thank you for your excellent work on this library. It's very much a boon to the Go ecosystem.

**Pull request merged**